src/modules/did/did.processor.ts

```diff
@@ -23,16 +39,30 @@ export class DIDProcessor {
   @Process(ADD_DID_DOC_QUEUE_NAME)
   public async processDIDDocumentAddition(job: Job<string>) {
     this.logger.debug(`processing cache add for ${job.data}`);
-    await this.didService.addCachedDocument(job.data);
+    const doc = await this.didService.addCachedDocument(job.data);
+
+    await this.pinClaims(doc.service.map((s) => s.serviceEndpoint));
   }
 
   @Process(UPDATE_DID_DOC_QUEUE_NAME)
   public async processDIDDocumentRefresh(job: Job<string>) {
     this.logger.debug(`processing cache refresh for ${job.data}`);
-    if (this.configService.get('DID_SYNC_MODE_FULL') === 'true') {
-      await this.didService.addCachedDocument(job.data, true);
+    let doc: DIDDocumentEntity;
+    if (this.config.get('DID_SYNC_MODE_FULL') === 'true') {
+      doc = await this.didService.addCachedDocument(job.data, true);
     } else {
-      await this.didService.incrementalRefreshCachedDocument(job.data);
+      doc = await this.didService.incrementalRefreshCachedDocument(job.data);
+    }
+
+    await this.pinClaims(doc.service.map((s) => s.serviceEndpoint));
+  }
+
+  private async pinClaims(cids: string[]) {
```

Comment:
What do you think of adding this documentation
```suggestion
  /**
   * This method migrates claims by retrieving from one DidStore and pinning to another
   * It was implemented for EW migration from Infura to EW hosted IPFS
   */
  private async pinClaims(cids: string[]) {
```